plugins/binanceExchange_ws.go

```diff
@@ -220,6 +327,20 @@ func subcribeBook(symbol string, state *mapEvents) (*stream, error) {
 
 }
 
+//ListenKey expires every 60 minutes
+func keepAliveStreamService(client *binance.Client, key string) {
+
+	for {
+		time.Sleep(time.Minute * 50)
+		err := client.NewKeepaliveUserStreamService().ListenKey(key).Do(context.Background())
+
+		if err != nil {
+			log.Printf("Error keepAliveStreamService %v\n", err)
+			panic(err)
```

Comment:
We should find a way to not panic here. Sorry didn’t see this earlier.
Panics are very bad in kelp for many reasons and I can go into detail if interested.